File: suffix_Array/DistinctSubstrings.java
import java.util.Scanner;
import java.util.TreeSet;

public class DistinctSubstrings {
    public static void main(String[] CSECU) {
        System.out.println("Aloha Bro: ");
        Scanner Arnab = new Scanner(System.in);
        System.out.print("Enter the String: ");
        String str= Arnab.next();
        System.out.println("Distinct Substrings Count: "+findDistinctSub(str));
        Arnab.close();
    }
    public static int findDistinctSub(String str){
        TreeSet<String> tree = new TreeSet<>();
        for(int a=0;a<str.length();a++){
            tree.add(str.substring(a));
        }

        int count=1;
        String text2=tree.first();

        for(String text: tree){
            if(text!=text2){
                count++;
            }
            text2=text;
        }
        return count;
    }
}

File: suffix_Array/Kotlin/DistinctSubstrings.kt
fun findDistinctSub(str: String): Int {
    val tree = sortedSetOf<String>()
    for (a in str.indices) {
        tree.add(str.substring(a))
    }

    var count = 1
    var text2 = tree.first()

    for (text in tree) {
        if (text != text2) {
            count++
        }
        text2 = text
    }
    return count
}

fun main() {
    println("Aloha Bro: ")
    print("Enter the String: ")
    val str = readLine() ?: ""
    println("Distinct Substrings Count: ${findDistinctSub(str)}")
} 